dotnet/src/SemanticKernel/AI/AIException.cs

```diff
@@ -76,6 +76,11 @@ public enum ErrorCodes
     /// </summary>
     public ErrorCodes ErrorCode { get; set; }
 
+    /// <summary>
+    /// The time to wait before retrying the request.
+    /// </summary>
+    public TimeSpan? RetryAfter { get; set; }
```

Comment:
strange approach, relying on an exception to drive business logic...

Reply:
It's actually an example in the Polly repository that refers to the Azure Client SDK doing this:
https://github.com/App-vNext/Polly/wiki/Retry#retryafter-when-the-response-specifies-how-long-to-wait
https://github.com/App-vNext/Polly/issues/247#issuecomment-438335852

Reply:
Agreed, felt odd at first when I was considering it, and then I found this... So I went for it.